src/main/java/build/buildfarm/cas/cfc/CASFileCache.java

```diff
@@ -247,6 +243,13 @@ public SettableFuture<Long> load(Digest digest) {
   private final transient Entry header = new SentinelEntry();
   private volatile long unreferencedEntryCount = 0;
 
+  private static final int MAX_ENTRIES = 10000;
+  private static final Cache<String, DigestFunction.Value> digest_to_function_map =
```

Comment:
why we need to introduce this cache?

Reply:
Without it this change actually requires a quite huge refactoring of changing many chaining functions, the corresponding interfaces and as a result also all other implementations of those. This seems to be the cleanest solution to me.